clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1364,6 +1364,10 @@ public void handleResponse(AbstractResponse response) {
             } else if (error == Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED ||
                     error == Errors.CLUSTER_AUTHORIZATION_FAILED) {
                 fatalError(error.exception());
+            } else if (error == Errors.INVALID_PRODUCER_EPOCH || error == Errors.PRODUCER_FENCED) {
+                // For older versions, we could still receive INVALID_PRODUCER_EPOCH from transaction coordinator.
```

Comment:
nit: ditto here, `We could still receive INVALID_PRODUCER_EPOCH from old versioned transaction coordinator.`